src/messages/store/publish.ts

```diff
@@ -68,7 +73,7 @@ export async function Publish({
         APIServer,
         inlineRequested,
         sync,
-        fileObject,
+        fileObject as Blob | File | undefined,
```

Comment:
This does not seem right to me, I'm not sure that the type assertion is the way to go. 
Can you provide some context as of why you do this

Reply:
So above we have `RequireOnlyOne` that makes sure that at least fileHash or fileObject is given. The assertion comes out of the need that if we have a fileObject, we read its content as a Buffer and then put it into a File or Blob object, depending on what it originally was. We simply drop the possibility of a Buffer being passed, which `createAndSendStoreMessage` does not accept anymore.

Reply:
Then maybe we should separate the method into two different functions, one for IPFS cid and the other one for binary payload. This will allow us to get rid of the big overlapping type assertions

Reply:
We definitely should rework the SDK at some point, but not right now tbh. ^^'

As long as it works in the tests and in the Browser, I'm happy